ambry-api/src/main/java/com/github/ambry/compression/Compression.java

```diff
@@ -25,17 +27,17 @@
  * Shared buffer APIs (caller management memory).
  *   Compression example:
  *   {@code
- *     byte[] originalData = ...;
- *     int compressedBufferSize = Compression.getCompressBufferSize(originalData.length);
- *     byte[] compressedBuffer = new byte[compressedBufferSize];
+ *     ByteBuffer originalData = ...;
```

Comment:
What was the motivation to go from byte array to a ByteBuffer ?

Reply:
nvm, i read the commit description

Reply:
ByteBuffer is more generic than byte[].  Byte[] is always Heap while ByteBuffer supports both Heap and Direct memory.  The output of Netty (incoming request) is Direct memory.  If compression takes Heap, Direct memory must be copied to Heap first.  Similarly, Netty prefers Direct memory when sending responses.  If Heap is sent to Netty, Netty makes a copy of the buffer before sending.  Also, when calling native code (compression in case this), native code only works with direct memory.  If we call with Heap like byte[], JVM either pin or copy the byte[] to direct memory.  Pinning is unpredictable and uncontrollable by ambry, so if ambry passes direct memory, we know it does not copy.